### ISSUE TITLE
PPAPI - Couldn't load plugin on E1.2.0 and Windows 

### ISSUE THREAD

**Issue by kjpou1** [electron/electron#5808]
* Electron version: 1.2.0
* Operating system: Surface Pro 4 - Windows 10

I have created a small sample program based on what was found here https://github.com/electron/electron/issues/3794.

The sample is found here:  
[electronpepper.zip](https://github.com/electron/electron/files/291142/electronpepper.zip)

Can not seem to get this to work at all in Electron 1.2.0 even though it works in chrome.

![chrome](https://cloud.githubusercontent.com/assets/640466/15675391/26ca9410-2742-11e6-9624-3516e3ff8ee6.PNG)

This is what happens running in Electron:

![electron](https://cloud.githubusercontent.com/assets/640466/15675435/583aeff4-2742-11e6-9eba-a6089e084957.PNG)

The zip file provided holds all the files to run the sample:

The hello.dll was created by:

- set NACL_SDK_ROOT=[sdk install path]\nacl_sdk\pepper_canary
- cd [path where electronpepper unzipped]
- make TOOLCHAIN=win

 

The following should be output:
```  CXX win/Release/main.o
  LINK win/Release/hello_unstripped.dll
   Creating library ./win/Release/hello_unstripped.lib and object ./win/Release/hello_unstripped.exp
  COPY win/Release/hello.dll```

This was all done from a Visual Studio 2013 x86 Native Tools Command Prompt.

To run in chrome from :

> [path to chrome]\chrome.exe --register-pepper-plugins="[unzipped path]\pepper\win\Release\hello.dll;application/x-ppapi-hello" file:///[unzipped path]\pepper\index.html

To test in Electron with electron-prebuilt:

> [unzipped path]electron.exe .\

Any help and/or advice would be greatly appreciated on what is happening here.

**Comment by paulcbetts** [comment]
What does electron print with --enable-logging?

**Comment by kjpou1** [comment]
I did ```set ELECTRON_ENABLE_LOGGING=true``` as per http://electron.atom.io/docs/api/environment-variables/ but nothing prints to the console.  Neither in the command prompt nor in the dev tools console window.

**Comment by deepak1556** [comment]
I am not sure about the logging problem but from the example you have plugin path as 

```
var ppapiPath = __dirname + 'win\\Release\\hello.dll';
```

have you made sure it loads from the right path ?

**Comment by kjpou1** [comment]
Ok, I am now a little further.  I started a new command window to start fresh, set the environment variable as above and hard coded the same path as used when running from chrome as stated above and am finally getting this in the command window:

Here is the error:

```[6380:0601/060041:ERROR:ppapi_thread.cc(310)] Failed to load Pepper module from C:\Xamarin\Pepper\pepper\win\Release\hello.dll (error: 193)```

So there is at least something happening with the logging now.  Why would there be an error with the ppapi_thread when it works in chrome?

The loading of the command line switch is basically the first thing that is done:

```
'use strict';
const electron = require('electron');
const app = electron.app;
const BrowserWindow = electron.BrowserWindow;
const path = require('path');

app.commandLine.appendSwitch('register-pepper-plugins', 'C:\\Xamarin\\Pepper\\pepper\\win\\Release\\hello.dll;application/x-ppapi-hello');
```

**Comment by enlight** [comment]
Which version of Chrome did you manage to load the plugin with?

**Comment by zcbenz** [comment]
Does the architecture of the plugin matches Electron (x86 or x64)?

**Comment by kjpou1** [comment]
I have been testing with two versions.  One installed on the windows machine and one I built from source.  One from source was for my testing to go the extra mile to cover all bases before posting here.

Here is the version that loads with Chrome that is installed on the windows machine with the Hello tab has the plugin loaded.  I can post that screen shot as well if needed.

"C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" --register-pepper-plugins="C:\Xamarin\Pepper\pepper\win\Release\hello.dll;application/x-ppapi-hello" file:///C:\Xamarin\Pepper\pepper\index.html

![chrome installed](https://cloud.githubusercontent.com/assets/640466/15698640/df8a21ee-27c6-11e6-9308-e919ea9d21b4.PNG)


Chromium built from source with the Hello tab and the plugin loaded:

c:\Xamarin\chromiumdesktop\src\out\Debug\chrome.exe --register-pepper-plugins="C:\Xamarin\Pepper\pepper\win\Release\hello.dll;application/x-ppapi-hello" file:///C:\Xamarin\Pepper\pepper\index.html

![chromium](https://cloud.githubusercontent.com/assets/640466/15698644/e4bddafc-27c6-11e6-94f8-1ac953383edc.PNG)

**Comment by paulcbetts** [comment]
What's the output of `process.arch`

**Comment by kjpou1** [comment]
Yes it was the architecture!!  I really thought I had the architecture set to 32 bit.  Must have changed things when I updated to 1.0 version and then the update to 1.2.0.  Sorry about that.

Just for documentation purposes here for anyone else running into this later I did the following:

```npm install --arch=ia32 electron-prebuilt -g```

When running with 32 bit architecture the plugin then runs fine

![electronpluginloaded](https://cloud.githubusercontent.com/assets/640466/15701677/105d790a-27dc-11e6-8b62-294d36cf7ebd.PNG)

Also attached a full working example:

[electronpepperia32.zip](https://github.com/electron/electron/files/292864/electronpepperia32.zip)

Thanks again.
